mqr/views.py

```diff
@@ -3,11 +3,50 @@
 from rest_framework import generics, permissions
 from rest_framework.response import Response
 
+from .models import MqrStrata
+from .serializers import MqrStrataSerializer
+
+arms = ["ARM", "RCM", "BCM", "RCM_BCM", "RCM_SMS"]
+
 
 class RandomArmView(generics.GenericAPIView):
     permission_classes = (permissions.IsAuthenticated,)
 
     def get(self, request):
         # TODO: change this to the strata sequential assignment
-        random_arm = random.choice(["ARM", "RCM", "BCM", "RCM_BCM", "RCM_SMS"])
+        random_arm = random.choice(arms)
         return Response({"random_arm": random_arm})
+
+
+class RandomStrataArmView(generics.GenericAPIView):
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = MqrStrataSerializer
+
+    def post(self, request):
+        serializer = MqrStrataSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        province = serializer.validated_data.get("province")
+        weeks_pregnant = serializer.validated_data.get("weeks_pregnant")
+        age = serializer.validated_data.get("age")
```

Comment:
Remember these two have buckets:
Weeks pregnant (3 ranges: 16-20 weeks, 21-25 weeks, 26-30 weeks)
Age (2 ranges: 18-30, 31+)

So before creating the record put them in buckets